pallets/parachain-staking/src/lib.rs

```diff
@@ -2012,19 +2045,49 @@ pub mod pallet {
 			Ok(())
 		}
 
-		fn kick_delegator(delegation: &StakeOf<T>, collator: &T::AccountId) -> DispatchResult {
+		/// Check for remaining delegations of the delegator which has been
+		/// removed from the given collator.
+		///
+		/// Returns the removed delegators address and
+		/// * Either the updated delegator state if other delegations are still
+		///   remaining
+		/// * Or `None`, signalling the delegator state should be cleared once
+		///   the transaction cannot fail anymore.
+		fn kick_delegator(
```

Comment:
S: Should we rename this perhaps to something like `prep_kick_delegator()` or similar? It is not making any storage changes anymore, so it is not in fact kicking a delegator.

Reply:
1db5f45